pkg/scheduler/util/utils.go

```diff
@@ -140,3 +140,11 @@ func MoreImportantPod(pod1, pod2 interface{}) bool {
 	}
 	return GetPodStartTime(pod1.(*v1.Pod)).Before(GetPodStartTime(pod2.(*v1.Pod)))
 }
+
+// NonBlockingSend sends val to the provided channel without blocking on the receiver.
+func NonBlockingSend(ch chan<- interface{}, val interface{}) {
```

Comment:
reminder: using `interface{}` here could double the time cost when calling this function because of typecasting even if the channel is empty.

```go
package main

import "testing"

func BenchmarkNonBlockingSend(b *testing.B) {
	iCh := make(chan interface{}, b.N)
	for i := 0; i < b.N; i++ {
		err := "score plugin scheduling error"
		NonBlockingSend(iCh, err)
	}
}

func BenchmarkNonBlockingSendError(b *testing.B) {
	iCh := make(chan string, b.N)
	for i := 0; i < b.N; i++ {
		err := "score plugin scheduling error"
		NonBlockingSendError(iCh, err)
	}
}

func NonBlockingSend(ch chan<- interface{}, val interface{}) {
	ch <- val
}

func NonBlockingSendError(ch chan<- string, err string) {
	ch <- err
}

$  go test -bench  .
goos: darwin
goarch: amd64
pkg: code.byted.org/zuoshuqi
BenchmarkNonBlockingSend-4        	20000000	        73.1 ns/op
BenchmarkNonBlockingSendError-4   	50000000	        34.2 ns/op
PASS 
```

We could use an interface channel here but use this function in other places should be careful.

Reply:
Thanks, good point. I agree, we should restrict using this to error reporting only. I will rename the function to indicate that it is for sending errors only, and so we can share the same implementation here and in other places including the PR you are working on.